src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import styled from '@emotion/styled';
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { useCallback } from 'react';
+import { Spinner } from '../spinner';
+import { CheckboxIcon } from './CheckboxIcon';
+
+interface Props {
+  checked: boolean | 'indeterminate';
+  className?: string;
+
+  label?: string;
+  ariaLabel?: string;
+  title?: string;
+  id?: string;
+
+  isDisabled?: boolean;
+  isLoading?: boolean;
+
+  onCheck: (checked: boolean | 'indeterminate', id?: string) => void;
+  onFocus?: VoidFunction;
+}
+
+export function Checkbox(props: Readonly<Props>) {
+  const { checked, className, label, ariaLabel, title, id, isDisabled, isLoading = false } = props;
+  const { onCheck, onFocus } = props;
+
+  const handleChange = useCallback(
+    (checked: boolean | 'indeterminate') => {
+      if (!isDisabled && !isLoading) {
+        onCheck(checked, id);
+      }
+    },
+    [isDisabled, isLoading, onCheck, id],
+  );
+
+  return (
+    <CheckboxContainer
+      aria-disabled={isDisabled}
+      as={label ? 'label' : 'span'}
+      className={className}>
+      <Spinner isLoading={isLoading}>
+        <CheckboxRoot
+          aria-disabled={isDisabled}
+          aria-label={ariaLabel ?? title}
+          checked={checked}
+          id={id}
+          onCheckedChange={handleChange}
+          onFocus={onFocus}
+          title={title}>
+          <CheckboxIndicator>
+            <CheckboxIcon checked={checked} />
+          </CheckboxIndicator>
+        </CheckboxRoot>
+      </Spinner>
+      {label && <Label>{label}</Label>}
+    </CheckboxContainer>
+  );
+}
+
+const CheckboxContainer = styled.span`
+  display: inline-flex;
+  align-items: center;
+  vertical-align: text-bottom;
+
+  &[aria-disabled='true'] {
+    color: var(--echoes-color-text-disabled);
+    pointer-events: none;
+  }
```

Comment:
We should also add cursor on hover

Reply:
I was wondering about that since the radix-ui implem and examples didn't have the hand cursor.

But [reading a bit about it](https://medium.com/simple-human/buttons-shouldnt-have-a-hand-cursor-b11e99ca374b) it seems like the hand cursor is actually for links only and doesn't mean "interactive". So I would keep it like this.

Reply:
And maybe we should also review our other usages of the cursor:pointer by the way.